chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+(e.g. GPS coordinates can be translated into distances). 
+However, if sensitive information is strictly needed for analysis,
+all the tasks described in this chapter must be performed in a secure environment.
 
-\section{Cleaning and constructing derivative datasets}
+% Unique ID and data entry corrections ---------------------------------------------
+There are two main cases when the raw data will be modified during data cleaning.
+The first one is when there are duplicated entries in the data.
+Ensuring that observations are uniquely and fully identified\sidenote{\url{https://dimewiki.worldbank.org/wiki/ID\_Variable\_Properties}}
+is possibly the most important step in data cleaning
+(as anyone who ever tried to merge data sets that are not uniquely identified knows).
+Modern survey tools create unique observation identifiers.
+That, however, is not the same as having a unique ID variable for each individual in the sample.
+You want to make sure the data set has a unique ID variable
+that can be cross-referenced with other records, such as the Master Data Set\sidenote{\url{https://dimewiki.worldbank.org/wiki/Master\_Data\_Set}}
+and other rounds of data collection.
+\texttt{ieduplicates} and \texttt{iecompdup}, 
+two Stata commands included in the \texttt{iefieldkit} 
+package,\index{iefieldkit}
+create an automated workflow to identify, correct and document
+occurrences of duplicate entries. 
 
-Once the de-identified dataset is created, data must be \textbf{cleaned}.\sidenote{
-\url{https://dimewiki.worldbank.org/wiki/Data_Cleaning}}
-\index{data cleaning}
-This process does not involve the creation of any new data fields.
-Cleaning can be a long process: this is the phase at which
-you obtain an extensive understanding of the contents and structure
-of the data you collected in the field.
-You should use this time to understand the types of responses collected,
-both within each survey question and across respondents.
-Understanding this structure will make it possible to do analysis.
+Looking for duplicated entries is usually part of data quality monitoring,
+as is the only other reason to change the raw data during cleaning:
+correcting mistakes in data entry.
+During data quality monitoring, you will inevitably encounter data entry mistakes,
+such as typos and inconsistent values.
+If you don't, you are probably not doing a very good job at looking for them.
+These mistakes should be fixed in the cleaned data set,
+and you should keep a careful record of how they were identified,
+and how the correct value was obtained.
 
-There are a number of tasks involved in cleaning data.
+% Data description ------------------------------------------------------------------
+Note that if you are using secondary data, 
+the tasks described above can likely be skipped.
+However, the last step of data cleaning, describing the data,
+will probably still be necessary.
```

Comment:
I don't know about using the word _describing_ here. I had to read a couple of times to understand myself. Maybe the sentence needs to be re-organized, as the first times I read this I kept reading it as if it was a list where _data cleaning_ was the first item, and _describing the data_ was the second item, not that the latter was a part of the first.

One quick fix would be _However, the last step of data cleaning — describing the data — will probably still be necessary.

Reply:
Change in c80a095:
>Note that if you are using secondary data, the tasks described above can likely be skipped.
>The last step of data cleaning, however, will most likely still be necessary.
>It consists of describing the data, so that its users have all the information needed to interact with it.

Reply:
"annotating"?

Reply:
Changed to "annotate" in 968ba4c. I was trying to avoid using "documenting" so that people don't think that this is all that is needed for documentation. But maybe elaborating more is preferable?

Reply:
What about "labelling and annotating"? I think annotating is correct but it would have made my list of necessarily difficult words. And it is more than just labelling. So that is why I think both should be used, _labelling_ for clarity, _annotating_ to make sure it includes enough. I did that edit in b595afd67ce53c4c252eaf30159c43dcd843ae81. Feel free to revert if you do not like it.

Reply:
hahaha. fair enough. that seems good to me. marking this conversation as resolved.